File: client-app/src/app/models/note.ts
export interface note {
    id: string;
    firstName: string;
    secondName: string;
    thridName?: string;
    phoneNumber: string;
    address?: string;
    description?: string;
    isImportant: boolean;
    creator: string;
    creationDate: Date | null;

}

export class note implements note {
    constructor(init?: noteFormValues){
        Object.assign(this, init)
    }
}

export class noteFormValues{
    id?: string = undefined;
    firstName: string = '';
    secondName: string = '';
    thridName?: string = '';
    phoneNumber: string = '';
    address?: string = ''
    description?: string = '';
    isImportant: boolean = false;
    creationDate: Date | null = null;
    constructor (note?: note){
        if (note){
            this.id = note.id;
            this.firstName = note.firstName;
            this.secondName = note.secondName;
            this.thridName = note.thridName;
            this.phoneNumber = note.phoneNumber;
            this.address = note.address;
            this.description = note.description;
            this.isImportant = note.isImportant;
            this.creationDate = note.creationDate;
        }
    }
}

File: Domain/Note.cs
namespace Domain;
public class Note
{
    public Guid Id { get; set; }
    public string FirstName { get; set; }
    public string SecondName { get; set; }
    public string ThridName { get; set; }
    public string PhoneNumber { get; set; }
    public string Address { get; set; }
    public string Description { get; set; }
    public bool IsImportant { get; set; }
    public DateTime CreationDate { get; set; }
    public AppUser Creator { get; set; }
}
